1_fetch/src/moving_window_functions.R

```diff
@@ -0,0 +1,139 @@
+
+calc_moving_window_metrics<-function(site_num, window_length,increment, min_yrs_in_window,clean_daily_flow,yearType, 
+                                     drainArea_tab, NE_probs, digits = 3 ){
+  ##subset clean_daily_flow, create a vector of starting years for a moving window analysis, and then
+  ##call the calcFDC function for each starting year.
+  data <- clean_daily_flow %>%
+    filter(site_no == site_num)
+  
+  
+  start_yrs<- seq(from = min(data$year_val), 
+                  to= (max(data$year_val) - window_length+1),
+                  by = increment)
+
+  ##remove any starting years which would result in a window with no data (if there are large data gaps). 
+  df_screen<-bind_rows(map(start_yrs,window_screen,data))
+  start_yrs<-start_yrs[start_yrs %in% df_screen$start_yr[which(df_screen$years_in_window > min_yrs_in_window)]==TRUE]
+  
+  
+  map_out<-map(start_yrs,calc_FDC_subset, 
+               window_length, site_num=site_num, 
+               clean_daily_flow=data,
+               yearType = yearType,
+               drainArea_tab=drainArea_tab,
+               NE_probs=NE_probs,digits=3)
+  data_out<-bind_rows(map_out)
+ 
+  return(data_out)
+}
+
+
+plot_trend_summary<-function(moving_window_metrics,outdir){
+  #normalize metrics and remove NAs (when there is only 1 moving window
+  #for a site,the sd will be 0)
+  df_norm<-moving_window_metrics %>%
+    group_by(site_num,indice) %>%
+    mutate(norm = (statistic - mean(statistic))/sd(statistic))%>%
+    mutate(indice_grp=word(indice,start=1,sep="_") )%>%
+    filter(!is.na(norm))
+  
+  indice_grp<-unique(df_norm$indice_grp) 
+  ##map over indice_grp and produce a plot file for each group, with all the quantiles 
+  map_out<-map(indice_grp,make_summary_plot,
+          data=df_norm,
+          outdir= outdir)%>%
+    unlist()
+
+  return(map_out)
+}#end function
+
+
+make_summary_plot<- function(grp,data, outdir){
+  df_plot<-data %>%
+    filter(indice_grp ==grp)
+  
+  p1<-ggplot(df_plot,aes(start_Year,norm)) +geom_point(size = .7,alpha = .2)+
+    geom_smooth()+
+    facet_wrap(~indice, ncol=3)
+  
+  filepath <- file.path(outdir, paste0("moving_window_summary_", grp, ".png"))
+  
+  ggsave(filename=filepath,
+         plot= p1)
+  return(filepath)
+}
+
+
+
+
+calc_FDC_subset<- function(start_yr, window_length,site_num, clean_daily_flow, yearType, 
+                           drainArea_tab, NE_probs, digits = 3 ){
+
+ ##subset the clean_daily_flow data
+
+  sub_clean_daily_flow<- clean_daily_flow %>%
+    filter((year_val >= start_yr) &(year_val < (start_yr + window_length)))
+  
+  df_FDC<- calc_FDCmetrics(site_num, 
```

Comment:
Just noting that I changed the return format of this function in my open PR. I can add an option to return in this format if that's simpler to work with. The `year_start` argument is also new to allow for non-water year seasons. So there will be some merge conflicts to resolve between these 2 PRs